etcd/v3/kv_etcd.go

```diff
@@ -803,43 +807,63 @@ func (et *etcdKV) watchStart(
 	if waitIndex != 0 {
 		opts = append(opts, e.WithRev(int64(waitIndex+1)))
 	}
+	session, err := concurrency.NewSession(et.kvClient, concurrency.WithTTL(defaultSessionTimeout))
```

Comment:
just to clarify, with this change in a single node etcd, session should terminate right ? (ignoring the reconnection issue ?)